File: src/main/java/com/example/ssauc/user/order/service/OrderService.java
package com.example.ssauc.user.order.service;

import com.example.ssauc.user.login.entity.Users;
import com.example.ssauc.user.login.repository.UsersRepository;
import com.example.ssauc.user.product.entity.Product;
import com.example.ssauc.user.product.repository.ProductRepository;
import lombok.AllArgsConstructor;
import org.springframework.stereotype.Service;

@Service
@AllArgsConstructor
public class OrderService {

    private ProductRepository productRepository;
    private UsersRepository usersRepository;

    public Product getProductById(Long productId) {
        return productRepository.findById(productId).orElseThrow();
    }

    public Users getUserById(Long userId) {
        return usersRepository.findById(userId).orElseThrow();
    }


}

File: src/main/java/com/example/ssauc/user/order/service/OrderService.kt
package com.example.ssauc.user.order.service

import com.example.ssauc.user.login.entity.Users
import com.example.ssauc.user.login.repository.UsersRepository
import com.example.ssauc.user.product.entity.Product
import com.example.ssauc.user.product.repository.ProductRepository
import lombok.AllArgsConstructor
import org.springframework.stereotype.Service

@Service
@AllArgsConstructor
class OrderService {
    private val productRepository: ProductRepository? = null
    private val usersRepository: UsersRepository? = null

    fun getProductById(productId: Long): Product {
        return productRepository!!.findById(productId).orElseThrow()
    }

    fun getUserById(userId: Long): Users {
        return usersRepository!!.findById(userId).orElseThrow()!!
    }
}
